### PR TITLE
Fix: PdfFileMerger can NOT handle encrypted PDFs

### PR DESCRIPTION
**Pull request by gaborpelesz**
## Problem 1:
The merger cannot handle decrypted PDF files when the PDF is provided via a PdfFileReader object, althought the logic is (somewhat) implemented.

### Understanding the problem and a solution:
The code below in theory handles the transfer of the decryption key of the original object, however, the previous line completely overwrites the original file object which eliminates its decryption key attribute.
```python
...
fileobj = filecontent
if hasattr(fileobj, '_decryption_key'):
    decryption_key = fileobj._decryption_key
```
As a solution the attribute extraction needs to be placed before the overwriting procedure.
```python
...
if hasattr(fileobj, '_decryption_key'):
    decryption_key = fileobj._decryption_key
fileobj = filecontent
```

## Problem 2:
Although if you give the merger a PDF encrypted with an empty password, it handles it well UNTIL you specify the pages explicitly as a tuple.

### Understanding the problem and a solution:
That is because when no pages were provided OR the pages were provided with a `PageRange` object, the method `getNumPages()` runs both times to handle the specific case.
```python
if pages == None:
    pages = (0, pdfr.getNumPages())
elif isinstance(pages, PageRange):
    pages = pages.indices(pdfr.getNumPages())
```
The `getNumPages()` method (interestingly) tries a decryption with an empty password that is why it works in those cases.

A solution is to try explicitly a decryption with an empty password if the PDF is encrypted and no decryption key was provided.
```python
if decryption_key is not None:
    pdfr._decryption_key = decryption_key
    elif pdfr.isEncrypted:
        # if the file is encrypted at least try 
        # the decryption with an empty password
        pdfr.decrypt('')
```

### CODE REVIEW

File: PyPDF2/merger.py [hunk 1]
```diff
@@ -139,6 +139,10 @@ def merge(self, position, fileobj, bookmark=None, pages=None, import_bookmarks=T
         pdfr = PdfFileReader(fileobj, strict=self.strict, overwriteWarnings=self.overwriteWarnings)
         if decryption_key is not None:
             pdfr._decryption_key = decryption_key
+        elif pdfr.isEncrypted:
+            # if the file is encrypted at least try
```
Comment (MartinThoma) [inline PyPDF2/merger.py:143]:
```suggestion
            # if the file is encrypted at least try
```

Comment (MartinThoma) [inline PyPDF2/merger.py:143]:
```suggestion
            # if the file is encrypted at least try
```

**Comment by MartinThoma** [comment]
Thank you very much for your contribution!

The core part of your PR was already merged in another context. For this reason, I'll close the PR.

There is still the empty password fallback. I will not merge this for the moment as this behavior is unexpected.

**Comment by gaborpelesz** [comment]
@MartinThoma Hey, this was a long time ago. Although I remember that I had many PDFs which I could open with any PDF reader without an issue, however, I was unable to open them with PyPDF2 because it thought that it was encrypted with a password. The solution was to just decrypt with an empty password. Cheers!

**Pull request closed without merging**